File: french/fr_phonetrack.php
<?php
if (isset($_GET['phone']))
	echo json_encode(french($_GET['phone']));

function french($number) {
    $handle = fopen("french_ranges.csv", "r");
    $dataArray = [];
    if ($handle) {
        while (($phone_range = fgets($handle)) !== false) {
            if (is_inrange($number, explode(",", $phone_range)[1], explode(",", $phone_range)[2])) {
                $dataArray = explode(",", $phone_range);
                array_push($dataArray, find_provider_name(explode(",", $phone_range)[3]));
            }
        }

        fclose($handle);
    } else {
        echo "Can't open the file";
    }
	return $dataArray;
}


function find_provider_name($code) {
    $handle = fopen("french_provider.csv", "r");
    if ($handle) {
        while (($line = fgets($handle)) !== false) {
            if (explode(",", $line)[0] == $code){
                return explode(",", $line)[1];
            }
        }

        fclose($handle);
    } else {
        echo "Can't open the file";
    } 
    return "PROVIDER_NOT_FOUND";
}

function is_inrange($phone, $rangesta, $rangesto) {
    for ($i = 0; $i != strlen($phone); $i++) {
        if ($phone[$i] < $rangesta[$i] or $phone[$i] > $rangesto[$i]) {
            return 0;
        }
    }
    return 1;
}

File: french/fr_phonetrack.py
import sys
import os

path = os.path.dirname(os.path.realpath(__file__))
provider_file = open(path + "/french_provider.csv", "r")
ranges_file = open(path + "/french_ranges.csv", "r")

def main():
    french(sys.argv[1])

def french(number: str):
    for phone_range in ranges_file:
        try:
            if is_inrange(number, phone_range.split(",")[1], phone_range.split(",")[2]):
                print("Phone range founded:", phone_range, end="")
                print("Provider:", find_provider_name(phone_range.split(",")[3]), end="")
        except:
            pass

def find_provider_name(code: str):
    for provider in provider_file:
        if provider.split(",")[0] == code:
            return provider.split(",")[1]
    return "PROVIDER_NOT_FOUND"

def is_inrange(phone, rangesta, rangesto):
    for digit in range(len(phone)):
        if int(phone[digit]) < int(rangesta[digit]) or int(phone[digit]) > int(rangesto[digit]):
            return 0
    return 1
if __name__ == "__main__":
    main()
